UKHomeOffice/sgmr-service
# SPCR-436 Do not allow dates like 31/2 to be entered

### AC / Description of changes

The issue
- if you entered 31 2 2023, it was being translated to 3/3/2023 (this is what js does)

What we wanted
- preserve what the user entered and error if `30/2`, `31/2`, `31/4`, `31/6`, `31/9`, `31/11`


### To Test

On departure page
- enter a date of 31 2 2023
- complete the other fields
- submit
- > this should error as date is invalid
- enter a date of 02 02 2023
- > this should be accepted as date is valid
- enter 2 2 2023
- this should be accepted as date is valid
- enter 32 1 2023
- > this should error as date is invalid
- enter 1 13 2023
- > this should error as date is invalid



### Notes

---

**Remember to merge to feature branches not `master`.**
